server/segment_rpc.go

```diff
@@ -328,6 +330,18 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64)
 		return nil, err
 	}
 
+	ratPrice, _ := ratPriceInfo(sess.OrchestratorInfo.PriceInfo)
+	faceVal := new(big.Int).SetBytes(sess.OrchestratorInfo.GetTicketParams().GetFaceValue())
```

Comment:
If we move the log statement into [this code block](https://github.com/livepeer/go-livepeer/blob/master/server/segment_rpc.go#L609) in `genPayment()` then we can avoid additional conversation steps and exporting `maxWinProb` from the `pm` package.

```
batch.TicketParams.Recipient
batch.TicketParams.FaceValue
batch.Tickets()[0].WinProbRat()
```

`WinProbRat()` is only defined on `pm.Ticket` right now, but might make sense to do something like:

```
type ticketParams struct {
   // All fields of TicketParams besides Seed
}

func (tp *ticketParams) WinProbRat() *big.Rat {}

// Embed ticketParams in TicketParams to expose WinProbRat()
// Embed ticketParams in Ticket to expose WinProbRat()
// Then in the above example we could do batch.TicketParams.WinProbRat()
```

Reply:
This would get annoying pretty quickly because of not being able to use promoted fields (fields on the embedded struct) in composite literals (eg everytime we call `&Ticket{...} `) so I would be okay of just using 0 based index on Tickets here instead of having to specifically set the fields seperately when creating tickets. 

To me this represents more of a bigger annoyance for me personally in go, embedding works well for methods but not for fields within a type when using composite literals. Perhaps we should default to using constructors only in this case and consider using composite literals with embedded types as an anti-pattern. 

WDYT ?

Reply:
Ah I see what you're saying - yeah updating usage of the `Ticket` constructor literal would be annoying.

Other options to avoid re-constructing the list of tickets (via `batch.Tickets()`) include:
1. Define `WinProbRat()` on `TicketParams` - the impl of `Ticket.WinProbRat()` could be extracted into an internal `winProbRat()` helper that is re-used in these two methods. Then we would just call `batch.WinProbRat()` for the log statement
2. Export a `WinProbRat()` helper that accepts a `big.Int` as a param

IMO Option 1 seems like it would read a bit better (due to the context around the type that the method is being called on).

Reply:
fixed in 5549ab8